Assets/Scripts/GameAI/Navigation/GroundedNavmeshNavigator.cs

```diff
@@ -40,6 +40,19 @@ public override void CheckIfPathNeedsToBeRegenerated()
             {
                 if (Vector3.Distance(navigationTarget.transform.position, lastKnownTargetPos) > NavigatorSettings.pathRefreshDistanceThreshold)
                 {
+                    //If the target has moved far enough away from their previous position, generate a new path.
+                    GeneratePathToTarget();
+                }
+            }
+        }
+
+        public override void CheckIfWaypointIsObstructed()
+        {
+            if (isActivelyGeneratingPath == true && navigationTarget != null)
```

Comment:
Is there a reason that you're explicitly comparing the boolean here with "== true"? I don't care stylistically. I'm just curious.

Reply:
Stylistic choice, I find that it makes the code more readable by making bools immediately recognizable.